crates/interpreter/src/valid.rs

```diff
@@ -645,6 +645,68 @@ impl<'a, 'm> Expr<'a, 'm> {
                 let t = self.context.table(x)?.item;
                 self.pops([ValType::I32, t.into(), ValType::I32][..].into())?;
             }
+            #[cfg(feature = "threads")]
+            AtomicNotify(m) => {
+                check(m.align == 2)?;
+                self.pops([ValType::I32, ValType::I32][..].into())?;
+                self.push(OpdType::I32);
+            }
+            #[cfg(feature = "threads")]
+            AtomicWait(n, m) => {
+                self.check_aligned(m, n.into())?;
+                self.pops([ValType::I32, NumType::i(n).into(), ValType::I64][..].into())?;
+                self.push(OpdType::I32);
+            }
+            #[cfg(feature = "threads")]
+            AtomicFence => (),
+            #[cfg(feature = "threads")]
+            IAtomicLoad(n, m) => {
+                self.check_aligned(m, n.into())?;
+                self.load(NumType::i(n), n.into(), m)?
+            }
+            #[cfg(feature = "threads")]
+            IAtomicLoad_(b, _, m) => {
+                self.check_aligned(m, b.into())?;
+                self.load(NumType::i(b.into()), b.into(), m)?;
+            }
+            #[cfg(feature = "threads")]
+            IAtomicStore(n, m) => {
+                self.check_aligned(m, n.into())?;
+                self.store(NumType::i(n), n.into(), m)?;
+            }
+            #[cfg(feature = "threads")]
+            IAtomicStore_(b, m) => {
+                self.check_aligned(m, b.into())?;
+                self.store(NumType::i(b.into()), b.into(), m)?;
+            }
+            #[cfg(feature = "threads")]
+            AtomicOp(n, syntax::AtomicOp::Cmpxchg, m) => {
```

Comment:
We import all the syntax in scope, so this shouldn't be needed
```suggestion
            AtomicOp(n, AtomicOp::Cmpxchg, m) => {
```

Reply:
I see. We don't have this issue with `IBinOp` and the other ops because the behavior is always the same for all ops.

Let's keep it with `syntax::AtomicOp` then.